src/analysis/ConfigurationsExt.cpp

```diff
@@ -304,6 +306,209 @@ namespace espressopp {
+              Real3D& vel = cit->velocity();
+              Int3D& img = cit->image();
+              Real3D L = system.bc->getBoxL();
+
+              coordinates[3*i]   = pos[0] + img[0] * L[0];
+              coordinates[3*i+1] = pos[1] + img[1] * L[1];
+              coordinates[3*i+2] = pos[2] + img[2] * L[2];
+
+              velocities[3*i]   = vel[0];
+              velocities[3*i+1] = vel[1];
+              velocities[3*i+2] = vel[2];
+
+              i++;
+            }
+          }
+          else{
+            for(CellListIterator cit(realCells); !cit.isDone(); ++cit) {
+
+              ids[i] = cit->id();
+              types[i] = cit->type(); // type of particle added
+
+              Real3D& pos = cit->position();
+              Real3D& vel = cit->velocity();
+
+              coordinates[3*i]   = pos[0];
+              coordinates[3*i+1] = pos[1];
+              coordinates[3*i+2] = pos[2];
+
+              velocities[3*i]   = vel[0];
+              velocities[3*i+1] = vel[1];
+              velocities[3*i+2] = vel[2];
+
+              i++;
+            }
+          }
+
+          if (i != myN) {
+            LOG4ESPP_ERROR(logger, "mismatch for number of local particles");
+          }
+
+          // each proc sends its data to master process
+
+          if (system.comm->rank() == 0) {
+
+             ConfigurationExtPtr config = make_shared<ConfigurationExt> (); //totalN
+
+             // root process collects data from all processors and sets it
+
+             int nproc = system.comm->size();
+
+             for (int iproc = 0; iproc < nproc; iproc++) {
+
+               int nother;
+
+               if (iproc) {
+
+                  int nIds, nCoords, nVelocs;   // number of received values
+                  //int tmp;
+                  size_t nTypes;
+
+                  boost::mpi::request req;
+                  boost::mpi::status  stat;
+
+                  LOG4ESPP_DEBUG(logger, "receive tags from " << iproc);
+
+                  req = system.comm->irecv<int>(iproc, DEFAULT_TAG, ids, maxN);
+                  system.comm->send(iproc, DEFAULT_TAG, 0);
+                  stat = req.wait();
+                  nIds = *stat.count<int>();
+
+                  req = system.comm->irecv<size_t>(iproc, DEFAULT_TAG, types, maxN);
+                  system.comm->send(iproc, DEFAULT_TAG, 0);
+                  stat = req.wait();
+                  nTypes = *stat.count<size_t>();
+
+                  req = system.comm->irecv<real>(iproc, DEFAULT_TAG, coordinates, 3*maxN);
+                  system.comm->send(iproc, DEFAULT_TAG, 0);
+                  stat = req.wait();
+                  nCoords = *stat.count<real>();
+
+                  // make sure to have 3 coordinate values for each id
+
+                  if (nCoords != 3 * nIds) {
+                    LOG4ESPP_ERROR(logger, "serious error collecting data, got " <<
+                                  nIds << " ids, but " << nCoords << " coordinates");
```

Comment:
shouldn't you stop the execution by throwing exception or MPI_Abort or exit(1), that serious error?